ViewControllers/SearchViewController.swift

```diff
@@ -0,0 +1,123 @@
+        tableView.backgroundColor = UIColor(named: "backgroundColor")
+        tableView.register(SearchTableViewCell.self)
+        tableView.rowHeight = 124
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+    
+    override func loadView() {
+        super.loadView()
+        view.backgroundColor = UIColor(named: "backgroundColor")
+        
+        title = "Search Movies"
+        navigationController?.navigationBar.barTintColor = UIColor(named: "backgroundColor")
+        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor: UIColor.white, .font: UIFont.systemFont(ofSize: 20, weight: .heavy)]
+        
+        navigationItem.searchController = searchController
+        searchController.searchBar.searchTextField.textColor = UIColor(named: "titleColor")!
+        
+        view.addSubview(tableView)
+        NSLayoutConstraint.activate([
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let notificationCenter = NotificationCenter.default
+        notificationCenter.addObserver(self, selector: #selector(makeRequest), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    @objc private func makeRequest() {
+        isSearching = true
+        updateSearchResults(for: searchController)
+    }
+}
+
+extension SearchViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        searchResults.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(SearchTableViewCell.self, for: indexPath)
+        
+        let movie = searchResults[indexPath.row]
+        cell.configure(imageURL: movie.posterPath, title: movie.title, reviewsScore: movie.voteAverage, popularity: movie.popularity)
+        
+        return cell
+    }
+}
+
+extension SearchViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let selectedMovieID = searchResults[indexPath.row].id
+        
+        let detailViewController = MovieDetailViewController(selectedMovieID: selectedMovieID)
+        let detailNavigationController = UINavigationController(rootViewController: detailViewController)
+        present(detailNavigationController, animated: true)
+    }
+}
+
+extension SearchViewController: UISearchResultsUpdating {
+    func updateSearchResults(for searchController: UISearchController) {
```

Comment:
Since we manually implemented the search handling methods I'd recommend leaving this method empty and moving your search code (code that is currently inside `updateSearchResults` to its own method, perhaps to the `makeRequest` method itself.

The method `updateSearchResults` is automatically called on any change of the search bar text content so you don't need to add observers or add targets to the search bar. The downside is that you will receive any change so you end up needing to have a debounce code here.

Reply:
The reason why I added observers was that updateSearchResults didn't trigger multiple searches. Once I entered one letter, search was launched and never repeated.